torch/package/importer.py

```diff
@@ -184,3 +184,11 @@ def import_module(self, module_name: str) -> ModuleType:
             raise last_err
         else:
             raise ModuleNotFoundError(module_name)
+
+    def whichmodule(self, obj: Any, name: str) -> str:
+        for importer in self._importers:
+            module_name = importer.whichmodule(obj, name)
```

Comment:
Might the importer implementation raise `ModuleNotFoundError`?

Reply:
`whichmodule` is not supposed to actually import any modules or have side effects into the import system, so I don't think so.